raidengine/raidengine.go

```diff
@@ -21,136 +16,80 @@ type MovementResult struct {
 		return err
 	}
+	tacticNames := viper.GetStringSlice(fmt.Sprintf("raids.%s.tactics", raidName))
+	tacticName := viper.GetString(fmt.Sprintf("raids.%s.tactics", raidName))
 
-	tacticsMultiple := fmt.Sprintf("raids.%s.tactics", raidName)
-	tacticSingular := fmt.Sprintf("raids.%s.tactic", raidName)
-
-	if viper.IsSet(tacticsMultiple) {
-		tactics := viper.GetStringSlice(tacticsMultiple)
-		for _, tactic := range tactics {
-			loggerName = fmt.Sprintf("%s-%s", raidName, tactic)
-			armory.SetLogger(loggerName)
-			viper.Set(tacticSingular, tactic)
-			newErr := ExecuteTactic(getStrikes(raidName, armory.GetTactics()))
-			if newErr != nil {
-				if err != nil {
-					err = fmt.Errorf("%s\n%s", err.Error(), newErr.Error())
-				} else {
-					err = newErr
-				}
-			}
+	if len(tacticNames) > 0 {
+		// Multiple tactics are specified
+		for _, tacticName := range tacticNames {
+			runTactic(raidName, tacticName, armory)
 		}
 		return err
-	} else if !viper.IsSet(tacticSingular) {
+	} else if tacticName != "" {
+		// Single tactic is specified, and multiple are NOT specified
+		err = runTactic(raidName, tacticName, armory)
+	} else {
 		err = fmt.Errorf("no tactics were specified in the config for the raid '%s'", raidName)
 		logger.Error(err.Error())
-		return err
 	}
-
-	// In case both 'tactics' and 'tactic' are set in the config for some ungodly reason:
-	loggerName := fmt.Sprintf("%s-%s", raidName, viper.GetString(tacticSingular))
-	armory.SetLogger(loggerName)
-	err = ExecuteTactic(getStrikes(raidName, armory.GetTactics()))
-	return err
+	return
 }
 
-// ExecuteTactic is used to execute a list of strikes provided by a Raid and customized by user config
-func ExecuteTactic(strikes []Strike) error {
-	closeHandler()
-
-	var attempts int
-	var successes int
-	var failures int
+// runTactic sets the tactic for a given raid, configures the logger, and executes the tactic.
+// If an error occurs during execution, it is returned.
+//
+// Parameters:
+//
+//	raidName: The name of the raid.
+//	tacticName: The name of the tactic to be executed.
+//	armory: The Armory instance used to set the logger and retrieve tactics.
+//
+// Returns:
+//
+//	err: An error if the tactic execution fails, otherwise nil.
+func runTactic(raidName string, tacticName string, armory Armory) (err error) {
+	loggerName = fmt.Sprintf("%s-%s", raidName, tacticName)
+	armory.SetLogger(loggerName)
 
-	tactic := &Tactic{
+	tactic := Tactic{
 		TacticName: loggerName,
-		StartTime:  time.Now().String(),
+		strikes:    armory.GetTactics()[tacticName],
 	}
 
-	for _, strike := range strikes {
-		attempts += 1
-		name, strikeResult := strike()
-		if strikeResult.Message == "" {
-			strikeResult.Message = "Strike did not return a result, and may still be under development."
-		}
-		if strikeResult.Passed {
-			successes += 1
-			logger.Info(strikeResult.Message)
+	newErr := tactic.Execute()
+	if newErr != nil {
+		if err != nil {
```

Comment:
dont think we need this if/else block, just 
```
err = fmt.Errorf("%s\n%s", err.Error(), newErr.Error())
```  
or
```
err = newErr
```
should do

Reply:
I'll push that up now